app/api/localStorage/index.js

```diff
@@ -27,127 +34,57 @@ export default class LocalStorageApi {
 
   // ========== Locale ========== //
 
-  getUserLocale = (): Promise<string> => new Promise((resolve, reject) => {
-    try {
-      const locale = localStorage.getItem(storageKeys.USER_LOCALE);
-      if (!locale) return resolve('');
-      resolve(locale);
-    } catch (error) {
-      return reject(error);
-    }
-  });
+  getUserLocale = (): Promise<string> => getLocalItem(storageKeys.USER_LOCALE);
 
-  setUserLocale = (locale: string): Promise<void> => new Promise((resolve, reject) => {
-    try {
-      localStorage.setItem(storageKeys.USER_LOCALE, locale);
-      resolve();
-    } catch (error) {
-      return reject(error);
-    }
-  });
+  setUserLocale = (locale: string): Promise<void> => setLocalItem(storageKeys.USER_LOCALE, locale);
 
-  unsetUserLocale = (): Promise<void> => new Promise((resolve) => {
-    try {
-      localStorage.removeItem(storageKeys.USER_LOCALE);
-      resolve();
-    } catch (error) {} // eslint-disable-line
-  });
+  unsetUserLocale = (): Promise<void> => removeLocalItem(storageKeys.USER_LOCALE);
 
   // ========== Terms of Use ========== //
 
-  getTermsOfUseAcceptance = (): Promise<boolean> => new Promise((resolve, reject) => {
-    try {
-      const accepted = localStorage.getItem(storageKeys.TERMS_OF_USE_ACCEPTANCE);
-      if (!accepted) return resolve(false);
-      resolve(JSON.parse(accepted));
-    } catch (error) {
-      return reject(error);
-    }
+  getTermsOfUseAcceptance = (): Promise<boolean> => getLocalItem(
+    storageKeys.TERMS_OF_USE_ACCEPTANCE
+  ).then((accepted) => {
+    if (accepted !== 'true') return false;
```

Comment:
this feels a little too verbose, maybe just something like this would be cleaner:

`return accepted !== 'true' ? false : true;`